payu/models/cesm_cmeps.py

```diff
@@ -156,15 +154,50 @@ def setup(self):
 
         self.runconfig.set("ALLCOMP_attributes", "start_type", start_type)
 
-        # Check pelayout makes sense
-        all_realms = self.realms + ["glc", "lnd"]
+        # check pelayout fits within requested cpucount
         cpucount = int(
             self.expt.config.get('ncpus', multiprocessing.cpu_count())
             )
+        all_realms = self.realms + ["glc", "lnd"]
         for realm in all_realms:
             ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
-            assert cpucount >= ntasks, "Insufficient cpus for the pelayout in nuopc.runconfig"
-        
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+            # rootpe is zero-based
+            if cpucount < (rootpe + ntasks*pestride):
+                raise ValueError(
+                    f"Insufficient cpus for the {realm} pelayout in nuopc.runconfig"
+                )
+
+        # check iolayout
+        for realm in self.realms:
+            # med and cpl names are both used in runconfig
+            if realm == "cpl": 
+                realm = "MED"
+            io_section = f"{realm.upper()}_modelio"
+            nc_type = self.runconfig.get(io_section, "pio_typename")
+            if nc_type == "netcdf4c":
+                raise ValueError(
+                    f"netcdf4c in {io_section} of nuopc.runconfig is deprecated, "
+                      "use netcdf4p"
+                )
+            else:
+                # if nc_type is netcdf, only one pe is needed
+                ioroot = int(self.runconfig.get(io_section, "pio_root"))
+                if cpucount < int(ioroot):
+                    raise ValueError(
+                        f"Insufficient cpus for the {io_section} ioroot pe in "
+                        "nuopc.runconfig"
+                    )
```

Comment:
Since `pio_root` was found to be non-functional in [#496 (comment)](https://github.com/payu-org/payu/pull/496/link), can it be removed? If `pio_root` does function, I am not sure why we need this additional check?

```suggestion

```

Reply:
My preference is we force the user to set pio_root correctly, rather than let the model change the value